crates/release_plz_core/src/git/backend.rs

```diff
@@ -170,6 +170,7 @@ pub struct GitLabMr {
     pub title: String,
     pub description: String,
     pub labels: Vec<String>,
+    pub remove_source_branch: bool,
```

Comment:
this struct is also Deserialize. 

we use the /merge_requests endpoint here:

https://github.com/release-plz/release-plz/blob/fd0a01b9a84a5ac849799636c51b139762113e6b/crates/release_plz_core/src/git/backend.rs#L385 

and we deserialize this struct here:

https://github.com/release-plz/release-plz/blob/fd0a01b9a84a5ac849799636c51b139762113e6b/crates/release_plz_core/src/git/backend.rs#L477

I don't think this field is returned. So imo instead of adding this field to this struct, we should add this field only when we create the gitlab pr